File: Try_GameEngine5/src/com/example/try_gameengine/framework/IChessPlayer.java
package com.example.try_gameengine.framework;

import java.util.List;

import android.graphics.Point;

public interface IChessPlayer extends IPlayer{
	public IChessPoint getChessPoint();
	public IChessPoint getPocessableMvoeChessPoint();
	boolean run(Point point, Point clickPoint, List<Point> allFreePoints);
}

File: Try_GameEngine5/src/com/example/try_gameengine/framework/IChessPlayer.kt
package com.example.try_gameengine.framework

import android.graphics.Point

interface IChessPlayer : IPlayer {
    fun getChessPoint(): IChessPoint?
    fun getPocessableMvoeChessPoint(): IChessPoint?
    fun run(point: Point?, clickPoint: Point?, allFreePoints: MutableList<Point?>?): Boolean
}
